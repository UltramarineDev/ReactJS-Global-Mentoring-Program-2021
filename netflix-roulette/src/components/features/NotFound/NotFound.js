import React from 'react';
import { Link } from 'react-router-dom';

import { wordings } from 'locales/wordings';
import Header from 'components/shared/Header/Header';
import Button from 'components/shared/button/Button';
import { buttonTypes, buttonSizes } from 'components/shared/constants';

import styles from './NotFound.module.scss';

const NotFound = () => (
  <>
    <Header />
    <div className={styles.container}>
      <h2 className={styles.title}>{wordings.page_not_found}</h2>
      <div className={styles.errorCode}>404</div>
      <Link to="/" className={styles.icon}>
        <Button
          label={wordings.go_back}
          type={buttonTypes.PRIMARY}
          size={buttonSizes.LARGE}
        />
      </Link>
    </div>
  </>
);

export default NotFound;
