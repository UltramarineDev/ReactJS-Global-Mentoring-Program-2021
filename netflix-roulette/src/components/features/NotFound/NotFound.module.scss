body {
  background-color: #232323;
}

.container {
  margin-top: 200px;
  left: 0;
  bottom: 0;
  text-align: center;
}

.title {
  font-size: 42px;
}

.errorCode {
  font-size: 250px;
}
