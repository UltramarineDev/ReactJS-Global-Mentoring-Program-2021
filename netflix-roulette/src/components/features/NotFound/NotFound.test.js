import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';

import NotFound from './NotFound';

describe('NotFound', () => {
  test('render NotFound', () => {
    const component = renderer
      .create(
        <MemoryRouter>
          <NotFound />
        </MemoryRouter>,
      )
      .toJSON();

    expect(component).toMatchSnapshot();
  });
});
