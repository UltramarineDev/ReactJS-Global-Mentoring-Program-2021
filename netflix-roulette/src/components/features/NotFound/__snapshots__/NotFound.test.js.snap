// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFound render NotFound 1`] = `
Array [
  <div
    className="header"
  >
    <span
      className="title"
    >
      <strong>
        netflix
      </strong>
      roulette
    </span>
  </div>,
  <div
    className="container"
  >
    <h2
      className="title"
    >
      Page Not Found
    </h2>
    <div
      className="errorCode"
    >
      404
    </div>
    <a
      className="icon"
      href="/"
      onClick={[Function]}
    >
      <a
        className="cancelButton largeSize"
        onClick={[Function]}
      >
        <span
          className="labelSearch"
        >
          Go back to home
        </span>
      </a>
    </a>
  </div>,
]
`;
